.env ファイル追加

cableguy123 opened the pull request:
.envファイル追加により build.gradle.kts 環境変数変更しました。

cableguy123 committed:
.env ファイル追加

Ryo0922ta commented:
https://discord.com/channels/1262794246604456059/1263157146263097465/1396144563256819752

このプルリクでは上記のやりとりでお話している
app/src/main/res/value/strings.xml
への変更が抜けている認識です。

この話題をお話している時に.envファイルに機密性の高い情報は.envファイルで読み込む形にしようというお話をしたと記憶してます。

ご確認お願いします。

cableguy123 commented:
@Ryo0922ta 
確認ありがとうございます。
app/src/main/res/values/strings.xmlで記載をしましたが、.envファイルよりこちら strings.xmlで記載したのは要らなくなるので
strings.xmlの変更盛れました。
修正します。

Ryo0922ta commented:
> @Ryo0922ta 確認ありがとうございます。 app/src/main/res/values/strings.xmlで記載をしましたが、.envファイルよりこちら strings.xmlで記載したのは要らなくなるので strings.xmlの変更盛れました。 修正します。


.envファイルに機密性の高い情報を記述し、変数経由でString.xmlにて.envファイルで記述した変数を読み出すコードを記述するため、String.xmlには.envファイルで記述した変数を読み出すコードを記述したものが
リモートにpushされて差分として表示される想定です。

認識に相違があればご連絡ください。

Ryo0922ta commented:
> > @Ryo0922ta 確認ありがとうございます。 app/src/main/res/values/strings.xmlで記載をしましたが、.envファイルよりこちら strings.xmlで記載したのは要らなくなるので strings.xmlの変更盛れました。 修正します。
> 
> .envファイルに機密性の高い情報を記述し、変数経由でString.xmlにて.envファイルで記述した変数を読み出すコードを記述するため、String.xmlには.envファイルで記述した変数を読み出すコードを記述したものが リモートにpushされて差分として表示される想定です。
> 
> 認識に相違があればご連絡ください。

cableguy123 committed:
【Issue】.env ファイル追加 #49
改修ファイル
build.gradle.kts
FirebaseModule.kt
RepositoryModule.kt
RoomModule.kt
削除ファイル
RepositoryProviderModule.kt

cableguy123 commented on app/build.gradle.kts:
```diff
@@ -24,20 +32,20 @@ android {
-            buildConfigField ( "String","BASE_URL","\" https://jp.co.we.travelbuddy/\"")
-            buildConfigField("String","DATA_SOURCE","\"firebase\"")
-            isMinifyEnabled = false
+//          signingConfig = signingConfigs.release
```
@Ryo0922ta 
リリース署名のため一旦こちらはコメントアウトしました。

cableguy123 commented on app/build.gradle.kts:
```diff
@@ -24,20 +32,20 @@ android {
-            isMinifyEnabled = false
+//          signingConfig = signingConfigs.release
+            isMinifyEnabled = true
+            isDebuggable = false
```
@Ryo0922ta 
デバックではないので falseに設定

cableguy123 commented on app/build.gradle.kts:
```diff
@@ -24,20 +32,20 @@ android {
             proguardFiles (
-                getDefaultProguardFile ( "proguard-android-optimize.txt" ),
-                "proguard-rules.pro"
+                getDefaultProguardFile ("proguard-android-optimize.txt"), "proguard-rules.pro"
```
@Ryo0922ta 
proguardFilesはコードを難読化ため必要ですので記載しました。

cableguy123 commented on app/build.gradle.kts:
```diff
@@ -24,20 +32,20 @@ android {
             )
         }
+        debug {
+            applicationIdSuffix = ".debug"
```
@Ryo0922ta 
パッケージ識別するため applicaitonIdSuffix = ".debug" 記載

cableguy123 commented on app/build.gradle.kts:
```diff
@@ -24,20 +32,20 @@ android {
+            applicationIdSuffix = ".debug"
+            isDebuggable = true
+            isMinifyEnabled = false
+            buildConfigField("String","WEB_CLIENT_ID", "\"${envProps["WEB_CLIENT_ID"]}\"")
```
@Ryo0922ta 
.env設定読み込み

cableguy123 commented on app/src/main/java/jp/co/we/travelbuddy/data/di/RoomModule.kt:
```diff
@@ -1,8 +1,5 @@
 package jp.co.we.travelbuddy.data.di
 
-// Room
-import androidx.room.Room
```
@Ryo0922ta 
使わない import文削除

cableguy123 commented on app/src/main/java/jp/co/we/travelbuddy/data/di/FirebaseModule.kt:
```diff
@@ -12,12 +12,15 @@ import dagger.hilt.components.SingletonComponent
     @Provides
     fun provideFirebaseStore() : FirebaseFirestore = FirebaseFirestore.getInstance()
-
-
```
@Ryo0922ta 
空白削除

Ryo0922ta commented:
@cableguy123 
コード改修箇所へのコメントありがとうございました。
改修対応確認しました。
マージお願いいたします。
